crates/bevy_ecs/src/system/query.rs

```diff
@@ -2558,9 +2558,9 @@ impl<'w, 'q, Q: QueryData, F: QueryFilter> From<&'q mut Query<'w, '_, Q, F>>
 /// [System parameter] that provides access to single entity's components, much like [`Query::single`]/[`Query::single_mut`].
 ///
 /// This [`SystemParam`](crate::system::SystemParam) fails validation if zero or more than one matching entity exists.
-/// This will cause the system to be skipped, according to the rules laid out in [`SystemParamValidationError`](crate::system::SystemParamValidationError).
+/// This will cause the system to fail, according to the rules laid out in [`SystemParamValidationError`](crate::system::SystemParamValidationError).
 ///
-/// Use [`Option<Single<D, F>>`] instead if zero or one matching entities can exist.
+/// Use [`When<Single<D, F>>`](crate::system::When) instead to skip the system when there are zero or multiple matching entities.
```

Comment:
Does this PR remove the ability to use `Option` like this? I think that should still work (since it's semantically different from `When`) and we should probably continue to document it here as well.

Reply:
Sure, I can update this to mention `Option` again!  That will still work after this PR.  

Note that the behavior changed slightly in #18766, though.  Previously it would yield `None` if there were zero entities and skip the system if there were multiple, but now it yields `None` even if there are multiple.  I removed the comment here because I read it as describing the old behavior, but updating it to describe the new behavior would be better!

Reply:
Makes sense! Thanks for the clarification!